src/ssim.jl

```diff
@@ -86,7 +86,7 @@ function _ssim_map(iqi::SSIM, x::GenericGrayImage, ref::GenericGrayImage, peakva
     C₁, C₂ = @. (peakval * K)^2
     C₃ = C₂/2
 
-    T = promote_type(floattype(eltype(ref)), floattype(eltype(x)))
+    T = promote_type(float(eltype(ref)), float(eltype(x)))
```

Comment:
I see the problems here, I used some latest exported function from `ColorTypes v0.8` and `FixedPointNumbers v0.6.1`. Unfortunately, not all packages in your local machine support them, so this's actually a compatibility issue.

In the meantime, `FixedpointNumbers.floattype` is used to automatically promote `N0f8` to `Float32` while still keep `AbstractFloat` not changed.

I think we could `using FixedPointNumbers`, and change `floattype` to `FixedPointNumbers.floattype`. (this function exists in FixedPointNumbers long time ago without exported)

Reply:
Just found out that there's a `float(x::Normed) = convert(floattype(x), x)` defined in `FixedPointNumbers`, so `float` behaves equivalently to `floattype`. Your change looks better than mine.

Reply:
lol